src/test/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/AlleleFilteringUnitTest.java

```diff
@@ -67,6 +62,76 @@ public void testNoNeedToFilter(){
+        haplotype.setEventMap(EventMap.fromHaplotype(haplotype, fullReferenceWithPadding, 0));
+
+        haplotype = new Haplotype("CATTCATG".getBytes(), false, 0, TextCigarCodec.decode("8M"));
+        haplotype.setGenomeLocation(new SimpleInterval("chr", 100, 108));
+        haplotypeList.add(haplotype);
+        haplotype.setEventMap(EventMap.fromHaplotype(haplotype, fullReferenceWithPadding, 0));
+
+        AlleleList<Haplotype> haplotypes = new IndexedAlleleList<>(haplotypeList);
+        SampleList samples = new IndexedSampleList(Arrays.asList(new String[]{"sm1", "sm2"}));
+
+        List<GATKRead> readList = new ArrayList<>(30);
+        Map<String, List<GATKRead>>ebs = new HashMap<>();
+        ebs.put("sm1", readList);
+
+        for (int i = 0 ; i < 30; i++) {
+            readList.add(ArtificialReadUtils.createArtificialRead("20M"));
+        }
+
+        List<GATKRead> readList2 = new ArrayList<>(30);
+        ebs.put("sm2", readList2);
+        for (int i = 0 ; i < 30; i++) {
+            readList2.add(ArtificialReadUtils.createArtificialRead("20M"));
+        }
+
+        double[][] values = {{0, 3, 0, 3, 0, 3, 0, 3, 0, 3, 0, 3, 0, 3, 0, 3, 0, 3, 0, 3, 0, 3, 0, 3, 0, 3, 0, 3, 0, 3},
+                {3, 0, 3, 0, 3, 0, 3, 0, 3, 0, 3, 0, 3, 0, 3, 0, 3, 0, 3, 0, 3, 0, 3, 0, 3, 0, 3, 0, 3, 0},
+                {0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0}
+        };
+        AlleleLikelihoods<GATKRead, Haplotype> lks = new AlleleLikelihoods<>(samples, haplotypes, ebs);
+        LikelihoodMatrix<GATKRead, Haplotype> lkm = lks.sampleMatrix(0);
+        for (int i = 0; i < lks.numberOfAlleles(); i++){
+            for (int j = 0 ; j < lkm.evidenceCount(); j++) {
+                lkm.set(i,j,values[i][j]);
+            }
+        }
+
+
+        double[][] values2 = {{0, 3, 0, 3, 0, 3, 0, 3, 0, 3, 0, 3, 0, 3, 0, 3, 0, 3, 0, 3, 0, 3, 0, 3, 0, 3, 0, 3, 0, 3},
+                {0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
+                {3, 0, 3, 0, 3, 0, 3, 0, 3, 0, 3, 0, 3, 0, 3, 0, 3, 0, 3, 0, 3, 0, 3, 0, 3, 0, 3, 0, 3, 0}
+        };
+        LikelihoodMatrix<GATKRead, Haplotype> lkm2 = lks.sampleMatrix(1);
+        for (int i = 0; i < lks.numberOfAlleles(); i++){
+            for (int j = 0 ; j < lkm2.evidenceCount(); j++) {
+                lkm2.set(i,j,values2[i][j]);
+            }
+        }
+
+        HaplotypeCallerArgumentCollection hcArgs = new HaplotypeCallerArgumentCollection();
+        HaplotypeCallerGenotypingEngine genotypingEngine = new HaplotypeCallerGenotypingEngine(hcArgs, samples, ! hcArgs.doNotRunPhysicalPhasing, false);
+
+
+        AlleleFiltering alleleFiltering = new AlleleFilteringHC(hcArgs, null, genotypingEngine);
+        AlleleLikelihoods<GATKRead, Haplotype> filtered_lks = alleleFiltering.filterAlleles(lks, 0, new HashSet<>());
+        Assert.assertEquals(filtered_lks.alleles(), lks.alleles());
```

Comment:
It looks like this checks that no alleles were filtered, but I'm confused how this tests the change that you added. Since it's taking the minimum PL across samples now (rather than the minimum PL from the first sample) then isn't it only possible to filter more alleles with the new version compared to the old version? Did this test fail with the original code before your change?

Reply:
See the comment below. Yes, this test failed in the old version. 
I am actually more conservative filtering the alleles now I think.(allele needs to be weak in both samples to be removed)